internal/controller/scheduler/limiter.go

```diff
@@ -93,7 +92,8 @@ func (l *MaxInFlightLimiter) Create(ctx context.Context, job *api.CommandJob) er
 		return nil
 	}
 
-	// Block until there's a token in the bucket.
+	// Block until there's a token in the bucket, or the job information becomes
```

Comment:
I think the intent of the code here doesn't really align with the comment

To block until something is stale is quite meaningless. Perhaps something like this wording could be used
```suggestion
	// Block until there's a token in the bucket, or cancel if the job information becomes stale.
```